src/components/FiltersModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import Modal from "react-modal";
+
+Modal.setAppElement("#root");
+
+type TFiltersModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  localFilters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  };
+  setLocalFilters: (filters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  }) => void;
+  onSubmit: (e: React.FormEvent) => void;
+};
+
+const FiltersModal: React.FC<TFiltersModalProps> = ({
+  isOpen,
+  onClose,
+  localFilters,
+  setLocalFilters,
+  onSubmit,
+}) => {
+  const toggleFilter = (key: keyof typeof localFilters) => {
+    setLocalFilters({
+      ...localFilters,
+      [key]: !localFilters[key],
+    });
+  };
+
+  const handleSearchQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLocalFilters({
+      ...localFilters,
+      searchQuery: e.target.value,
+    });
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onRequestClose={onClose}
+      contentLabel="Filter Options"
+      className="modal-content"
+      overlayClassName="modal-overlay"
+    >
+      <h2 className="modal-title">Filters & Search</h2>
+      <div className="filters-container">
+        {[
+          ["isHiddenGem", "Hidden gem"],
+          ["isOnView", "On view"],
+          ["iPublicDomain", "Public domain"],
+        ].map(([value, text]) => (
```

Comment:
Define this array outside of the rendering code, because then you can type it and avoid typecasting `value` everytime you use it:

```ts
const booleanFilterOptions: [MyFilter, string][] = […]
```

Also, this array is static – it always has the same, unchanging values. So, it doesn't need to be defined in the component.

Reply:
Also, by typing the list from the get-go TypeScript points out typos:

<img width="276" alt="Screenshot 2025-02-17 at 17 09 58" src="https://github.com/user-attachments/assets/feb29c93-66c4-4933-bdd0-95f4b29dc840" />


```suggestion
        {[
          ["isHiddenGem", "Hidden gem"],
          ["isOnView", "On view"],
          ["isPublicDomain", "Public domain"],
        ].map(([value, text]) => (
```